bb/ramfs.go

```diff
@@ -50,19 +50,41 @@ func sanity() {
 	}
 }
 
+// dirComponents takes a string and returns an array of strings,
+// such that we can create the directory records for intermediate
+// directories.
+func dirComponents(dir string) []string {
+	var dirlist []string
+	if filepath.Dir(dir) == "." {
+		return []string{}
+	}
+	for d := filepath.Dir(dir); d != "/"; d = filepath.Dir(d) {
+		dirlist = append([]string{d}, dirlist...)
+	}
+	dirlist = append(dirlist, dir)
+	return dirlist
+}
+
+// copyCommands takes a list of commands, generates the list of libs,
+// and creates cpio records, including directory records.
 func copyCommands(w cpio.Writer, cmds []string) {
+	debug("copyCommands: start with %v", cmds)
 	var recs []cpio.Record
 	libs, err := uroot.LddList(cmds)
```

Comment:
it does.
the ldd command pulls all that stuff in. I've been using it for a few days now.